### PR TITLE
[unit test] Add cmd/pkg/shell

### PR DESCRIPTION
**Pull request by masch**
Hi @HarryMichal!

I've created the first unit test using on cmd/pkg/shell. I've chosen only this package in order to get you opinion on the mechanism that I've proposed.
As we discussed on #473, I used [testify](https://github.com/stretchr/testify) library in order to get more easy and cleaner test cases.

Please, let know what you think and anything kind of suggestion will be very appreciate!

### CODE REVIEW

**Comment by masch** [comment]
Hi @HarryMichal !

I'm sorry to bother you again, but  did you have a change to take on look on the PR?
Is there anything I can do fix the build on the CI?

Thanks!

**Comment by HarryMichal** [comment]
I'm sorry for the delay, I've been busy somewhere else.

Overall I like this! I'd just pause merging for a little while until we land some PRs that are needed for a v0.0.91.

Thank you for working on this! I appreciate it a lot!

**Comment by masch** [comment]
No problem, great!

Is there anything another task to help?
How do you track issues in order to take one?

**Comment by HarryMichal** [comment]
Currently it is a bit chaotic (there's no roadmap currently). A great task right now would be to expand our tests (or to be more accurate, create them 😅). You started very well and if you don't mind writing tests then I'd personally welcome that a lot!

**Comment by masch** [comment]
Yes, sure. I will do more. Should I add them on the same PR o should i create another?
Which is the best way to get notified when there will be a good time on the roadmap? I don't want to bother you again if there is a channel or a system to get the status of it.

File: src/cmd/root.go [hunk 1]
```diff
@@ -380,5 +380,5 @@ func validateSubIDFile(path string) (bool, error) {
 		}
 	}
 
-	return false, fmt.Errorf("failed to find an entry for user %s in %p", currentUser.Username, path)
+	return false, fmt.Errorf("failed to find an entry for user %s in %s", currentUser.Username, path)
```
Comment (debarshiray) [inline src/cmd/root.go:383]: Could you please split this hunk out into a separate PR because it's related to the unit tests?

**Comment by HarryMichal** [comment]
> Thanks for working on this!
> 
> I haven't had a chance to actually run the tests. It would be helpful if you could mention how they are meant to be run.

You need to go to the src directory and run `go test ./...`. The 3 dots are important.

File: src/cmd/root.go (hunk 1 above)
Comment (masch) [reply src/cmd/root.go:383]: Yes, sure. I will remove it from this PR and I will add it on a new one.

**Comment by masch** [comment]
> I am curious why you are using _cmd/pkg/shell_? :) It should probably be just _pkg/shell_.
> 
> Also, why not call it _tests_ or something?

I don't know the rules that you follow, but you are right, `pkg\shell` is a better name. I will rename de commit message.

**Comment by masch** [comment]
> > Thanks for working on this!
> > I haven't had a chance to actually run the tests. It would be helpful if you could mention how they are meant to be run.
> 
> You need to go to the src directory and run `go test ./...`. The 3 dots are important.

Thank you for your mentioned, you are right.
Also, I always do on my projects this kind of task by makefiles. I love it, you can automate a lots of task.
I usually use a few task from this great repository template for go applications: https://github.com/golang-templates/seed#build

**Comment by HarryMichal** [comment]
> Thank you for your mentioned, you are right.
> Also, I always do on my projects this kind of task by makefiles. I love it, you can automate a lots of task.
> I usually use a few task from this great repository template for go applications: [golang-templates](https://github.com/golang-templates/seed#build)

We use `meson` + `ninja` over `make` in this project. Running the unit tests could certainly be wired into it.

**Comment by masch** [comment]
> > Thank you for your mentioned, you are right.
> > Also, I always do on my projects this kind of task by makefiles. I love it, you can automate a lots of task.
> > I usually use a few task from this great repository template for go applications: [golang-templates](https://github.com/golang-templates/seed#build)
> 
> We use `meson` + `ninja` over `make` in this project. Running the unit tests could certainly be wired into it.

Oh! I don't know nothing about those tools, I'll read some articles about them.
Do you have a Fedora example project that use them? It will be great to know how do you usually use them.

**Comment by HarryMichal** [comment]
You can find Meson's documentation [here](https://mesonbuild.com/). It is used a lot by GNOME projects (e.g., [Shell](https://gitlab.gnome.org/GNOME/gnome-shell/), [Contacts](https://gitlab.gnome.org/GNOME/gnome-contacts), [Foliate](https://github.com/johnfactotum/foliate) - strictly speaking not a GNOME project - and many others).

Generally, you mostly care about Meson, not about ninja.

File: src/cmd/root.go (hunk 1 above)
Comment (masch) [reply src/cmd/root.go:383]: I've created the [480](https://github.com/containers/toolbox/pull/480) PR  with the minor fix on it.

**Comment by masch** [comment]
One more question, I don't know how you usually manage the PR. Once I create a new one, Should I assign it to review to you guys?
Thanks!

**Comment by HarryMichal** [comment]
> One more question, I don't know how you usually manage the PR. Once I create a new one, Should I assign it to review to you guys?
> Thanks!

I can give you some pointers and fix some obvious errors (obvious enough for me to spot them :smile: ) but @debarshiray currently has the final say in what goes in or not.

**Comment by masch** [comment]
> > One more question, I don't know how you usually manage the PR. Once I create a new one, Should I assign it to review to you guys?
> > Thanks!
> 
> I can give you some pointers and fix some obvious errors (obvious enough for me to spot them smile ) but @debarshiray currently has the final say in what goes in or not.

Great! Thank you for your help, it is very helpful. 
Also I want to ask If I create a PR, should I live it without assigning a reviewer or should I assign to you guys?

**Comment by HarryMichal** [comment]
I wouldn't care about assigning it to anybody unless it takes us ages to respond.

**Comment by masch** [comment]
Hi @debarshiray, sorry to ping you.

I'm sorry for the delay, but I didn't notice that there was some changes on the `go.mod` files on master and it have generated some conflict. 

Please let me know what do you think and I will add more test on another PR.

**Comment by HarryMichal** [comment]
Hi @masch! I am terribly sorry for pushing your work aside for very long time. I took another look at this and took the liberty of rebasing the PR and updating it slightly.

I changed the name of the test cases variable (`tc` -> `testCases`), removed the casts to `string` since they are not needed because all the values are set to be `bytes` already and renamed `stdOutMock` to `outputMock` so that it can be used for both `stdout`/`stderr` mocking.

One more time, I apologize for not getting back to your work earlier. @debarshiray?

**Comment by HarryMichal** [comment]
I had a chat with @debarshiray on chat and I've got the green light to merge this. Thank you, @masch!!
